packages/webpack-import-meta-loader/webpack-import-meta-loader.js

```diff
@@ -24,9 +24,34 @@ module.exports = function(source) {
   const browserPath = toBrowserPath(relativePath);
 
   const fileName = this.resource.substring(this.resource.lastIndexOf(path.sep) + 1);
-  return source.replace(
-    regex,
-    () =>
-      `({ url: \`\${window.location.protocol}//\${window.location.host}/${browserPath}/${fileName}\` })`,
-  );
+
+  let found = false;
+  let rewrittenSource = source.replace(regex, () => {
+    found = true;
+    return `({ url: getAbsoluteUrl('${browserPath}/${fileName}') })`;
```

Comment:
can we make this behavior optional, based on a flag or only if a public path is set?

Reply:
Well, that's basically what the `getAbsoluteUrl` function is doing. The flag however is only available at runtime. In the first implementation I had this check (and thus the contents of `getAbsoluteUrl`) directly within the declaration of `({url: ... })`. But it was a big line with two ternary operators and a lot of duplication. I felt it would be more elegant to put it in a separate function to keep the code clean and consistent, but also to avoid to make the compiled code too big.

Reply:
As a reference, here is (more or less) the previous code: 
```javascript
    return `({ url: __webpack_public_path__ ? (__webpack_public_path__.indexOf('://') > 0 ? __webpack_public_path__ : \`\${window.location.protocol}//\${window.location.host}\${__webpack_public_path__}\` ) : \`\${window.location.protocol}//\${window.location.host}//\`) + \`\${browserPath}/\${fileName}\` })`;
```